mkosi/distributions/gentoo.py

```diff
@@ -27,41 +30,86 @@
 
 def invoke_emerge(
     state: MkosiState,
+    sysroot: Optional[Path] = None,
+    root: Optional[Path] = None,
+    bwrap_params: list[PathString] = [],
     pkgs: Sequence[str] = (),
     actions: Sequence[str] = (),
     opts: Sequence[str] = (),
+    env: dict[str, str] = {},
 ) -> None:
     jobs = os.cpu_count() or 1
+    if sysroot is not None:
+        target_root_mntp = "/tmp/mkosi-root"
```

Comment:
This should maybe be relative to the workspace directory.

Reply:
not sure, could you elaborate please?

Reply:
@behrmann could we mark this as resolved?

Reply:
Sorry, missed your question. My point was, why is this path hardcoded and not relative to `state.workspace`?

Reply:
it's just mount-point inside "/tmp" so it disappears after each `bwrap` call. is it worth making it relative to `state.workspace`?
also, any example where we do something like this somewhere else?

Reply:
I don't think we should scatter random mount points outside of the directories we directly control. At least don't hardcode it to `/tmp/`, but use `mkosi.util.tmp_dir()` and some function from the `tempfile` module so as not to make the path predictable.

Reply:
> I don't think we should scatter random mount points outside of the directories we directly control. At least don't hardcode it to `/tmp/`, but use `mkosi.util.tmp_dir()` and some function from the `tempfile` module so as not to make the path predictable.

how is that *outside of the directories we directly control*?

Reply:
that's a mount-point *inside* `/tmp`, bind-mounted as `tmpfs` again inside root tree of what we get from upstream (stage3) which we save under `cache_dir`. even stage3 is not tainted by this, so i'm not sure what you mean by mount point outside of the ddirectories we control.

marking as resolved, unresolve if need be.

Reply:
that is NOT host's `/tmp/`

Reply:
Applogies, I confused that, but maybe that's an indication to put a comment there to make the context for calling it this way clear.